### PR TITLE
S2 Rewrite with better API

### PR DESCRIPTION
**Pull request by justMaku**

### CODE REVIEW

File: Dependencies/S2 Geometry/S2Cell.m [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                                        j:j + (r[3] & 1)
+                      originalOrientation:originalOrientation
+                                 position:position + 3
+                              orientation:orientation ^ kPosToOrientation[3]];
+    }
+}
+
+- (instancetype)initWithIdentifier:(uint64)identifier {
+    self = [super init];
+
```
Comment (a2) [inline Dependencies/S2 Geometry/S2Cell.m]:
```objc
if (self) {
    self.identifier = identifier;
}
```

File: Dependencies/S2 Geometry/S2Types.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#ifndef S2Types_h
+#define S2Types_h
+
+typedef signed char         schar;
+typedef signed char         int8;
+typedef short               int16;
+typedef int                 int32;
+
+typedef unsigned char      uint8;
```
Comment (a2) [inline Dependencies/S2 Geometry/S2Types.h]: I think there are types like `uint8_t`, `uint16_t`, `uint32_t`, `uint64_t`, and so on.

File: Kanto/Network/Authentication/PokemonTrainerClubAuthenticator.swift [hunk 3]
```diff
@@ -26,7 +27,7 @@ public class PokemonTrainerClubAuthenticator: Authenticator  {
     
     public var token: String? = nil
     public var ticket: Ticket? = nil
-    
+
```
Comment (a2) [inline Kanto/Network/Authentication/PokemonTrainerClubAuthenticator.swift]: Remove trailing whitespace?

File: Dependencies/S2 Geometry/S2Cell.m (hunk 1 above)
Comment (justMaku) [inline Dependencies/S2 Geometry/S2Cell.m]: NSObject's init can't fail afaik.

**Pull request merged**